mlir/include/mlir/IR/MLIRContext.h

```diff
@@ -195,7 +195,12 @@ class MLIRContext {
 
   /// Return a sorted array containing the information about all registered
   /// operations.
-  ArrayRef<RegisteredOperationName> getRegisteredOperations();
+  SmallVector<RegisteredOperationName, 0> getRegisteredOperations();
```

Comment:
It's not clear to me why we can't just keep returning an ArrayRef?

A copy into a vector makes it quite expensive.

Reply:
Yes, I had this in my mind, with the latest change I have added a member variable so we can continue to use `ArrayRef` whilst using `transform`.